arch/x86/asm-offsets.c

```diff
@@ -33,4 +33,37 @@
 
 void __asm_offset_header(void) {
     OFFSETOF(usermode_private, percpu_t, usermode_private);
+
+    OFFSETOF(cpu_exc_error_code, cpu_exc_t, error_code);
+    OFFSETOF(cpu_exc_vector, cpu_exc_t, vector);
+    OFFSETOF(cpu_exc_ip, cpu_exc_t, _ASM_IP);
+    OFFSETOF(cpu_exc_cs, cpu_exc_t, cs);
+    OFFSETOF(cpu_exc_flags, cpu_exc_t, _ASM_FLAGS);
+    OFFSETOF(cpu_exc_sp, cpu_exc_t, _ASM_SP);
+    OFFSETOF(cpu_exc_ss, cpu_exc_t, ss);
+
+    OFFSETOF(cpu_regs_ax, cpu_regs_t, _ASM_AX);
+    OFFSETOF(cpu_regs_bx, cpu_regs_t, _ASM_BX);
+    OFFSETOF(cpu_regs_cx, cpu_regs_t, _ASM_CX);
+    OFFSETOF(cpu_regs_dx, cpu_regs_t, _ASM_DX);
+    OFFSETOF(cpu_regs_si, cpu_regs_t, _ASM_SI);
+    OFFSETOF(cpu_regs_di, cpu_regs_t, _ASM_DI);
+    OFFSETOF(cpu_regs_bp, cpu_regs_t, _ASM_BP);
+    OFFSETOF(cpu_regs_r8, cpu_regs_t, r8);
+    OFFSETOF(cpu_regs_r9, cpu_regs_t, r9);
+    OFFSETOF(cpu_regs_r10, cpu_regs_t, r10);
+    OFFSETOF(cpu_regs_r11, cpu_regs_t, r11);
+    OFFSETOF(cpu_regs_r12, cpu_regs_t, r12);
+    OFFSETOF(cpu_regs_r13, cpu_regs_t, r13);
+    OFFSETOF(cpu_regs_r14, cpu_regs_t, r14);
+    OFFSETOF(cpu_regs_r15, cpu_regs_t, r15);
```

Comment:
Should the 64 bit registers be wrapped in a `#ifdef __x86_64__` ... `#endif` block covering lines 52 to 59?
I mean, you already took care of making the code work for 32 bit by using the `_ASM_*` register name indirection. Would only make sense to guard the 64 bit registers as well.